packages/backend-for-frontend/src/services/clientService.ts

```diff
@@ -240,18 +241,21 @@ export function clientServiceBuilder(
         headers,
       });
 
-      const users = clientUsers.map(async (id) =>
-        toBffApiCompactUser(
-          await getSelfcareUserById(
-            selfcareUsersClient,
-            id,
-            selfcareId,
-            correlationId
-          ),
-          id
-        )
+      return await Promise.all(
+        clientUsers.map(async (id) => {
+          try {
+            const user = await getSelfcareUserById(
+              selfcareUsersClient,
+              id,
+              selfcareId,
+              correlationId
+            );
+            return toBffApiCompactUser(user, id);
+          } catch (error) {
+            throw userNotFound(id, selfcareId);
+          }
+        })
```

Comment:
```suggestion
            const user = await getSelfcareUserById(
              selfcareUsersClient,
              id,
              selfcareId,
              correlationId
            );
            return toBffApiCompactUser(user, id);
        })
```
I don't think we should throw `userNotFound` since we don't know what kind of error happend; IMHO we should remove the try/catch alltogether and let the error be propagated, i think it's fine returning `500` if the error comes from selfcare, I think we handle it like this in all places

Reply:
Ok, so it would just be caught by the makeApiProblem and returned as a generic API Problem - makes sense 👌